galsim/chromatic.py

```diff
@@ -173,6 +173,22 @@ def __init__(self, obj):
         self.wave_list = obj.wave_list
         self.deinterpolated = obj.deinterpolated
 
+    @property
+    def gsparams(self):
+        return self._obj.gsparams
+
+    def withGSParams(self, gsparams):
+        """Create a version of the current object with the given gsparams
+
+        Note: if this object wraps other objects (e.g. Convolution, Sum, Transformation, etc.)
+        those component objects will also have their gsparams updated to the new value.
+        """
+        if gsparams is self.gsparams: return self
```

Comment:
I was slightly worried that == would be too slow.  There are a fair number of places that we kind of gratuitously call `withGSParams` with things that are often the same actual object (but occasionally aren't, so it's not actually completely gratuitous).  So the `is` check is usually the relevant one.

And I think the time to do the `==` check is roughly the same time as the `GSParams.check` line below, so I'm not sure that there is much advantage in checking for equality (rather than identity).

Reply:
Fair enough.  I'm sure you're right that `gsp1 is gsp2` occurs much more often than `gsp1 is not gsp2 and gsp1 == gsp2`.  And I suppose a user could be careful to use id-identical gsps in the (probably uncommon?) situation where you'd want `withGSParams` to elide a copy.